### PR TITLE
modernize various tests

### PR DESCRIPTION
**Pull request by ajm188**
## Description

Some go1.22 changes and some `assert`/`require` changes.

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/15182
- https://github.com/vitessio/vitess/issues/15183

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by frouioui** [comment]
Adding `Backport to: release-19.0` as the branch has been updated to `go1.22` too.

File: go/vt/mysqlctl/mysqlctlproto/backup_test.go [hunk 1]
```diff
@@ -102,8 +102,6 @@ func TestBackupHandleToProto(t *testing.T) {
 	}
 
 	for _, tt := range tests {
-		tt := tt
```
Comment (frouioui) [inline go/vt/mysqlctl/mysqlctlproto/backup_test.go:105]: Nice

**Review by frouioui — APPROVED** [review]
LGTM!

File: .golangci.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ linters-settings:
     disable: # not supported when using Generics in 1.18
       - nilness
       - unusedwrite
+      - loopclosure # fixed in go1.22
```
Comment (deepthi) [inline .golangci.yml:14]: 💯

**Review by deepthi — APPROVED** [review]

**Comment by deepthi** [comment]
Are you making these changes by hand? We had an intern create a [tool](https://github.com/arvind-murty/test-rewriter) for this last year, but he only finished the `go/mysql` package during his term.

**Comment by ajm188** [comment]
I was doing these by hand but I'll check that out for my next pass!

**Comment by ajm188** [comment]
Added a link to the tool to https://github.com/vitessio/vitess/issues/15182

**Pull request merged**